packages/commonwealth/client/scripts/views/pages/Communities/TokensList/TokensList.tsx

```diff
@@ -0,0 +1,102 @@
+import { CWButton } from 'client/scripts/views/components/component_kit/new_designs/CWButton';
+import clsx from 'clsx';
+import { useFlag } from 'hooks/useFlag';
+import { navigateToCommunity, useCommonNavigate } from 'navigation/helpers';
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { useFetchTokensQuery } from 'state/api/tokens';
+import { CWText } from 'views/components/component_kit/cw_text';
+import CWCircleMultiplySpinner from 'views/components/component_kit/new_designs/CWCircleMultiplySpinner';
+import TokenCard from '../../../components/TokenCard';
+import './TokensList.scss';
+
+const TokensList = () => {
+  const navigate = useCommonNavigate();
+  const tokenizedCommunityEnabled = useFlag('tokenizedCommunity');
+
+  const {
+    data: tokensList,
+    isInitialLoading,
+    isFetchingNextPage,
+    hasNextPage,
+    fetchNextPage,
+  } = useFetchTokensQuery({
+    cursor: 1,
+    limit: 8,
+    enabled: tokenizedCommunityEnabled,
+  });
+  const tokens = (tokensList?.pages || []).flatMap((page) => page.results);
+
+  const handleFetchMoreTokens = () => {
```

Comment:
wondering how pagination will work for two lists in the same view (tokens  + communities)

Reply:
The communities pagination works on scroll (like before), while this one involves user to click "load more" button.